jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ContentSourceTransformer.java

```diff
@@ -31,11 +34,10 @@ public abstract class ContentSourceTransformer implements Content.Source
 {
     private static final Logger LOG = LoggerFactory.getLogger(ContentSourceTransformer.class);
 
+    private final AtomicReference<State> state = new AtomicReference<>(State.IDLE);
     private final Content.Source rawSource;
     private Content.Chunk rawChunk;
```

Comment:
`rawChunk` also needs to be volatile as there isn't always a good visibility-enforcing action. `fail()` for instance modifies this variable without enforcing its visibility.